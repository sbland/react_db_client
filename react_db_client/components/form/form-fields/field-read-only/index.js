export { FieldReadOnly } from './field-read-only';
