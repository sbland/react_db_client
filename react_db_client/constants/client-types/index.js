export { filterTypes, filterTypesDefaults } from './filter-types';

export {
  comparisons,
  stringComparisons,
  numberComparisons,
  dateComparisons,
} from './comparisons';

export {
  FilterObjectSimpleClass,
  FilterObjectClass,
} from './filter-object-class';
