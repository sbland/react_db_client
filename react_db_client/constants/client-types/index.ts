export * from './filter-types';

export * from './comparisons';

export { FilterObjectSimpleClass, FilterObjectClass, FilterOption } from './filter-object-class';

export * from './file';

export * from './uid';

export type { ILabelled } from './lib';

export type { IDocument } from './IDocument';

export type {
  TAsyncGetDocument,
  TAsyncGetDocuments,
  TAsyncPutDocument,
  TAsyncPostDocument,
  TAsyncDeleteDocument,
  TAsyncCopyDocument,
} from './api-queries';
