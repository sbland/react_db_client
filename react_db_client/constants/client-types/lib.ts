import { Uid } from "./uid";

export interface ILabelled<IdType extends Uid = Uid> {
  uid: IdType;
  label: string;
}
