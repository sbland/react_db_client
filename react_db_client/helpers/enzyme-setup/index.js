import { configure } from 'enzyme';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import Adapter from 'enzyme-adapter-react-16';


expect.extend({ toMatchDiffSnapshot });
configure({ adapter: new Adapter() });