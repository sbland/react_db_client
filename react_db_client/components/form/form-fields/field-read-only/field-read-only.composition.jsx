import React from 'react';
import { FieldReadOnly } from './field-read-only';
import { filterTypes } from '@react_db_client/constants.client-types';

export const BasicFieldReadOnly = () => (
  <FieldReadOnly
    value="hello from FieldReadOnly"
    unit="TEXT"
    type={filterTypes.text}
    options={[]}
  />
);
