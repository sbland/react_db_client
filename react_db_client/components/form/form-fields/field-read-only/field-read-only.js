import React from 'react';
import PropTypes from 'prop-types';
import { filterTypes } from '@samnbuk/react_db_client.constants.client-types';

const getSelectValue = (value, options) => {
  const val = options && options.find((o) => o.uid === value);
  return val ? val.label : null;
};

export const FieldReadOnly = ({ unit, value, type, options }) => {
  let val = value;
  if (type === filterTypes.select) val = getSelectValue(value, options);
  if (type === filterTypes.bool) val = value ? 'Yes' : 'no';
  if (typeof value == 'object') val == 'INVALID';
  return (
    <>
      <span>{val}</span>
      <span className="unitSpan">{unit && ` ${unit}`}</span>
    </>
  );
};

FieldReadOnly.propTypes = {
  unit: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
    PropTypes.object,
    PropTypes.array,
  ]),
  type: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      uid: PropTypes.string,
      label: PropTypes.string,
    })
  ),
};

FieldReadOnly.defaultProps = {
  unit: '',
  value: '',
  options: [],
};

export default FieldReadOnly;
