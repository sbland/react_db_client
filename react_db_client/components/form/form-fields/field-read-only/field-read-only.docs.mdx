---
description: 'A FieldReadOnly component.'
labels: ['form_field']
---

import { FieldReadOnly } from './field-read-only';
import { filterTypes } from '@react_db_client/constants.client-types';

## React Component for rendering read only fields

A basic div that renders some text

### Using props to customize the text

Modify the text to see it change live:

```js live
<FieldReadOnly
  value="hello from FieldReadOnly"
  unit="TEXT"
  type={filterTypes.text}
  options={{}}
/>
```
