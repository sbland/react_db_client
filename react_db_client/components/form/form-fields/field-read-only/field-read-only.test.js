import '@samnbuk/react_db_client.testing.enzyme-setup';

import React from 'react';
import { mount, shallow } from 'enzyme';
import { filterTypes } from '@react_db_client/constants.client-types';

import { FieldReadOnly } from './field-read-only';
import * as compositions from './field-read-only.composition';

const defaultProps = {
  unit: 'unit',
  value: 'a',
  type: filterTypes.text,
};

describe('FieldReadOnly', () => {
  beforeEach(() => {});
  test('Renders', () => {
    shallow(<FieldReadOnly {...defaultProps} />);
  });

  describe('Compositions', () => {
    Object.entries(compositions).forEach(([name, Composition]) => {
      test(name, () => {
        mount(<Composition />);
      });
    });
  });
  describe('shallow renders', () => {
    test('Matches Snapshot', () => {
      const component = shallow(<FieldReadOnly {...defaultProps} />);
      const tree = component.debug();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('Unit tests', () => {
    let component;
    beforeEach(() => {
      component = mount(<FieldReadOnly {...defaultProps} />);
    });
    test('should show value in span', () => {
      expect(component.text()).toEqual(`${defaultProps.value} ${defaultProps.unit}`);
    });
    test('should show option label if type is select', () => {
      const label = 'A';
      component = mount(
        <FieldReadOnly
          {...defaultProps}
          type={filterTypes.select}
          options={[{ uid: 'a', label }]}
        />
      );
      expect(component.text()).toEqual(`${label} ${defaultProps.unit}`);
    });
  });
});
