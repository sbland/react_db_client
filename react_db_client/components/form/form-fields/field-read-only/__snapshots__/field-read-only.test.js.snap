// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldReadOnly shallow renders Matches Snapshot 1`] = `
"<Fragment>
  <span>
    a
  </span>
  <span className=\\"unitSpan\\">
     unit
  </span>
</Fragment>"
`;
